File: frontend/src/api/navigation.ts
import request from './request'

export interface NavigationCategory {
  id?: number
  parent_id?: number | null
  name: string
  description?: string
  icon?: string
  sort: number
  children?: NavigationCategory[]
}

export interface Navigation {
  id?: number
  category_id?: number | null
  name: string
  url: string
  icon?: string
  description?: string
  sort: number
  status: number
  category?: NavigationCategory
}

export interface NavigationCategoryListParams {
  page?: number
  page_size?: number
  keyword?: string
}

export interface NavigationCategoryListResponse {
  list: NavigationCategory[]
  total: number
  page: number
  page_size: number
}

export interface NavigationListParams {
  page?: number
  page_size?: number
  keyword?: string
  category_id?: number
}

export interface NavigationListResponse {
  list: Navigation[]
  total: number
  page: number
  page_size: number
}

// 获取网站分类列表
// 如果带分页参数，返回分页数据
// 如果不带分页参数，返回树形结构
export const getNavigationCategoryList = (params?: NavigationCategoryListParams): Promise<NavigationCategoryListResponse | NavigationCategory[]> => {
  return request.get('/navigation-categories', { params })
}

// 获取网站分类详情
export const getNavigationCategoryById = (id: number): Promise<NavigationCategory> => {
  return request.get(`/navigation-categories/${id}`)
}

// 创建网站分类
export const createNavigationCategory = (data: NavigationCategory): Promise<NavigationCategory> => {
  return request.post('/navigation-categories', data)
}

// 更新网站分类
export const updateNavigationCategory = (id: number, data: Partial<NavigationCategory>): Promise<void> => {
  return request.put(`/navigation-categories/${id}`, data)
}

// 删除网站分类
export const deleteNavigationCategory = (id: number): Promise<void> => {
  return request.delete(`/navigation-categories/${id}`)
}

// 获取网站列表（分页）
export const getNavigationList = (params?: NavigationListParams): Promise<NavigationListResponse> => {
  return request.get('/navigations', { params })
}

// 获取网站详情
export const getNavigationById = (id: number): Promise<Navigation> => {
  return request.get(`/navigations/${id}`)
}

// 创建网站
export const createNavigation = (data: Navigation): Promise<Navigation> => {
  return request.post('/navigations', data)
}

// 更新网站
export const updateNavigation = (id: number, data: Partial<Navigation>): Promise<void> => {
  return request.put(`/navigations/${id}`, data)
}

// 删除网站
export const deleteNavigation = (id: number): Promise<void> => {
  return request.delete(`/navigations/${id}`)
}

File: backend/internal/model/navigation.go
package model

// Navigation 网站模型
type Navigation struct {
	BaseModel
	CategoryID  *uint               `gorm:"index:idx_nav_category_id;comment:所属分类ID" json:"category_id"`
	Category    *NavigationCategory `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Name        string              `gorm:"type:varchar(100);not null;comment:网站名称" json:"name"`
	URL         string              `gorm:"type:varchar(255);not null;comment:网站地址" json:"url"`
	Icon        string              `gorm:"type:varchar(50);comment:网站图标" json:"icon"`
	Description string              `gorm:"type:text;comment:描述" json:"description"`
	Sort        int                 `gorm:"type:int;default:0;index:idx_nav_sort;comment:排序" json:"sort"`
	Status      int                 `gorm:"type:tinyint;default:1;index:idx_nav_status;comment:状态 1-启用 0-禁用" json:"status"`
}

// TableName 指定表名
func (Navigation) TableName() string {
	return "navigations"
}
